build_tools/benchmarks/post_benchmarks_as_pr_comment.py

```diff
@@ -80,14 +72,19 @@ def get_git_total_commit_count(commit: str, verbose: bool = False) -> int:
   return int(count)
 
 
-def get_origin_tree_commit(distance: int, verbose: bool = False) -> str:
+def get_origin_tree_commit(base_branch: str,
+                           distance: int,
+                           verbose: bool = False) -> str:
   """Returns the hash for the commit with the given distance from top of the
   tree for the origin base branch."""
-  base_branch = get_required_env_var("BUILDKITE_PULL_REQUEST_BASE_BRANCH")
-  execute_cmd_and_get_output(
-      ['git', 'fetch', '--prune', '--', 'origin', base_branch],
-      cwd=THIS_DIRECTORY,
-      verbose=verbose)
+  # --no-recurse-submodules avoids git multiple configurations warnings on
+  # submodules when the current repo is at a merged commit generated by Github
+  # CI run.
+  cmds = [
+      'git', 'fetch', '--prune', '--no-recurse-submodules', '--', 'origin',
```

Comment:
This extra fetch shouldn't be necessary on GitHub actions if we just fetch with history in the first place. It also introduces an unnecessary dependency on when this was run. It's definitely not necessary to do it in a loop, since it should be fetching the same thing each time. In fact, it means that the loop can potentially return the same commit multiple times! We should really be querying vs the parent of the commit this was run against. GitHub by default checks out the merge commit of the PR into the base branch, so this is close, but the base branch could've advanced in the meantime. If we fetch the merge commit with history, we just need to walk the first parent of the current commit. If it's the PR head commit, then we should still have its merge base if we fetch with history that's not a problem, and it should be easy to find which commit it is assuming things are appropriately labeled. If they're not, we may need to explicitly fetch the base branch as well, but from there should just find the merge base.

If changing this creates a lot of complications with Buildkite, we can wait until after the transition, but I'd like to see this cleaned up.

Reply:
Fixed in #11692